walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,64 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+    List<TimeRange> relatedEvents = new ArrayList<TimeRange>();
+    Collection<TimeRange> answers = new ArrayList<TimeRange>();
+
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return answers;
+    }
+
+    for (Event event: events){
+        for (String attendee: attendees) {
+            if (event.getAttendees().contains(attendee)) {
+                relatedEvents.add(event.getWhen());
+                break;
+            }
+        }
+    }
```

Comment:
This is an example of something that can go into a helper function.

Reply:
so what I should do about this? move it to a child function?

Reply:
Yes, you can move it to a static helper function that takes in a list of events and attendees and  returns a list of related events